controllers/vspherevm_controller.go

```diff
@@ -351,6 +356,18 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 	return reconcile.Result{}, nil
 }
 
+func (r vmReconciler) isWaitingForStaticIPAllocation(ctx *context.VMContext) bool {
+	devices := ctx.VSphereVM.Spec.Network.Devices
+	for _, dev := range devices {
+		if !dev.DHCP4 && !dev.DHCP6 && len(dev.IPAddrs) == 0 {
```

Comment:
I'd rather have this as:
```
if dev.DHCP4 || dev.DHCP6 || len(dev.IPAddrs) != 0 {
       return false
}
```
easier to reason about

Reply:
This won't work if more than one device is waiting for static IP. We want to wait till all devices have the static IP.

Reply:
after giving some thoughts, this might be the more straight-forward way of thinking about this